frontend/src/app/main/ui/viewer/handoff/attributes/common.cljs

```diff
@@ -11,43 +11,69 @@
   (:require
    [rumext.alpha :as mf]
    [cuerdas.core :as str]
+   [okulary.core :as l]
+   [app.common.math :as mth]
    [app.util.dom :as dom]
    [app.util.i18n :refer [t] :as i18n]
    [app.util.color :as uc]
-   [app.common.math :as mth]
-   [app.main.ui.icons :as i]
    [app.util.code-gen :as cg]
    [app.util.webapi :as wapi]
+   [app.main.ui.icons :as i]
+   [app.main.store :as st]
    [app.main.ui.components.copy-button :refer [copy-button]]
    [app.main.ui.components.color-bullet :refer [color-bullet color-name]]))
 
+
+(def file-colors-ref
+  (l/derived (l/in [:viewer-data :file :colors]) st/state))
+
+(defn make-colors-library-ref [file-id]
+  (let [get-library
+        (fn [state]
+          (get-in state [:viewer-libraries file-id :data :colors]))]
+    #(l/derived get-library st/state)))
+
 (mf/defc color-row [{:keys [color format copy-data on-change-format]}]
-  (let [locale (mf/deref i18n/locale)]
+  (let [locale (mf/deref i18n/locale)
+
+        colors-library-ref (mf/use-memo
```

Comment:
Doesn't display anything when it's a color from the local library, instead of an external library.